### ISSUE TITLE
.so file not found

### ISSUE THREAD

**Issue by j-manu** [akihikodaki/cld3-ruby#37]
Getting this error after installation in Docker container
```
Could not open library '/usr/local/bundle/gems/cld3-3.5.0/lib/cld3/../../ext/cld3/libcld3.so': /usr/local/bundle/gems/cld3-3.5.0/lib/cld3/../../ext/cld3/libcld3.so: cannot open shared object file: No such file or directory (LoadError)
```

The `.so` file is present in `/usr/local/bundle/gems/cld3-3.5.0/lib/` but not `/usr/local/bundle/gems/cld3-3.5.0/ext/cld3/libcld3.so` 

Reproduction: https://github.com/debug-poc/cld3 
https://github.com/debug-poc/cld3/actions/runs/3781540490/jobs/6428505336

**Comment by j-manu** [comment]
This is due to https://github.com/rubygems/rubygems/issues/6205

**Comment by akihikodaki** [comment]
tl; dr: cld3-ruby won't support the latest RubyGems until RubyGems developers figure out a reliable way to get the path to the binary. I recommend to stay away of the latest RubyGems and Ruby, which includes the lastest RubyGems for now.

I authored commit 0ff7bb57a052fbdc346f28c1e1e9b097bfc606b6 but I decided *not* to merge this change. The problem of this change is that the binary won't be installed to the `lib` directory on Fedora and on Arch Linux (and potentially on other distributions as well). For details, see:
- https://src.fedoraproject.org/rpms/rubygems/c/6aa4b2f223987a0ea3fe06da1633058174ccf871
- https://github.com/archlinux/svntogit-community/blob/packages/rubygems/trunk/rubygems_stop_so_duplication.patch

Referring to the `lib` directory can also be problematic in the future as the binary in the directory is stated to be removed anyway: https://github.com/rubygems/rubygems/commit/e07dba09231a5182cab8a00094cbc261cd1ed0fc

Unfortunately the reality is that RubyGems introduced a disruptive change without realizing its nature and there is no good workaround. I hope the situation gets better soon; only a few days have passed since Ruby 3.2.0 release anyway so let's wait a bit longer.

**Comment by josh-m-sharpe** [comment]
Is there an active/open discussion against ruby gems somewhere?  I see that https://github.com/rubygems/rubygems/issues/6205 was linked, but is closed now.

**Comment by akihikodaki** [comment]
I don't think so. There is a PR to document an API useful to fix the issue but it's kind of stale and does not provide an exact procedure to resolve the path: https://github.com/rubygems/rubygems/pull/6218

**Comment by j-manu** [comment]
@akihikodaki Please consider releasing a version compatible with ruby 3.2 even if it is not optimal for all use cases. You can release it as prerelease gem so that it has to be opted in explicitly.

**Comment by akihikodaki** [comment]
Fixed with https://github.com/akihikodaki/cld3-ruby/commit/59f3ae9b5cecb88dda8ece28969c97ddfb0b85f4.

**Comment by j-manu** [comment]
@akihikodaki Thank you!!
